src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java

```diff
@@ -84,6 +87,14 @@ public static StructuredSyslogMessage fromString(
 
     private void  deserialize(final String stringMessage) {
 
+        // Check if the RFC5424 MSGID and STRUCTURED-DATA fields are empty.
+        // This avoids throwing an exception and also strips the "- - " from the message.
+        // See: https://github.com/Graylog2/graylog2-server/issues/1161
+        if (stringMessage.startsWith(EMPTY_STRUCTURED_DATA_PREFIX)) {

```

Comment:
A little late to the party, sorry.

Shouldn't we check if `message` is null or empty and that `structuredData` is an empty collection? Otherwise messages which start with "- - " intentionally will incorrectly get truncated.